xarray/coding/calendar_ops.py

```diff
@@ -204,7 +204,14 @@ def convert_calendar(
         out[dim] = new_times
 
         # Remove NaN that where put on invalid dates in target calendar
```

Comment:
Thanks @larsbuntemeyer !

It seems like we might want to fix it at the `where` call referred to here. The core issue is #1234. We could work around it by being careful about what's passed to `where`

Reply:
That's good to know, thanks @dcherian ! That seemed to cause some confusion indeed.